### ISSUE TITLE
Users with "No self-service" data access can't download report with filters

### ISSUE THREAD

**Issue by kanobi** [metabase/metabase#20868]
**Describe the bug**
Error happens when "no self-service" user tries to export question in csv/excel/json. Error only appears when the question is part of a dashboard with connected filter and user clicks on a question title.
Export works normally when opening the question directly from the collection.
For users with unrestricted data access, there's no error in any case.

**Logs**
```
{"via":[{"type":"clojure.lang.ExceptionInfo","message":"Invalid parameter: Card 1 does not have a template tag named nil.","data":{"type":"invalid-parameter","invalid-parameter":{"type":"date/all-options","value":"2018-03-05~2022-03-05","target":["dimension",["field",34,null]],"id":"f6dd6d27"},"allowed-parameters":null},"at":["metabase.query_processor.card$fn__68401$validate_card_parameters__68406$fn__68407","invoke","card.clj",160]}],"trace":[["metabase.query_processor.card$fn__68401$validate_card_parameters__68406$fn__68407","invoke","card.clj",160],["metabase.query_processor.card$fn__68401$validate_card_parameters__68406","invoke","card.clj",151],["metabase.query_processor.card$run_query_for_card_async","invokeStatic","card.clj",208],["metabase.query_processor.card$run_query_for_card_async","doInvoke","card.clj",169],["clojure.lang.RestFn","invoke","RestFn.java",826],["metabase.api.card$fn__71249","invokeStatic","card.clj",686],["metabase.api.card$fn__71249","invoke","card.clj",678],["compojure.core$wrap_response$fn__32273","invoke","core.clj",160],["compojure.core$wrap_route_middleware$fn__32257","invoke","core.clj",132],["compojure.core$wrap_route_info$fn__32262","invoke","core.clj",139],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",151],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",152],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",152],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",152],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",152],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",152],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",152],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285","invoke","core.clj",200],["metabase.server.middleware.auth$enforce_authentication$fn__65369","invoke","auth.clj",14],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285","invoke","core.clj",200],["compojure.core$make_context$handler__32313","invoke","core.clj",289],["compojure.core$make_context$fn__32317","invoke","core.clj",299],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$make_context$fn__32317","invoke","core.clj",300],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$make_context$fn__32317","invoke","core.clj",300],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$make_context$fn__32317","invoke","core.clj",300],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["metabase.api.routes$fn__76493$fn__76496","invoke","routes.clj",56],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285","invoke","core.clj",200],["clojure.lang.AFn","applyToHelper","AFn.java",160],["clojure.lang.AFn","applyTo","AFn.java",144],["clojure.core$apply","invokeStatic","core.clj",667],["clojure.core$apply","invoke","core.clj",662],["metabase.server.routes$fn__76638$fn__76639","doInvoke","routes.clj",57],["clojure.lang.RestFn","invoke","RestFn.java",436],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285","invoke","core.clj",200],["compojure.core$make_context$handler__32313","invoke","core.clj",289],["compojure.core$make_context$fn__32317","invoke","core.clj",299],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["compojure.core$wrap_route_matches$fn__32266","invoke","core.clj",153],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285$f__32286$respond_SINGLEQUOTE___32287","invoke","core.clj",197],["metabase.server.routes$fn__76626$fn__76628","invoke","routes.clj",41],["compojure.core$routes$fn__32285$f__32286","invoke","core.clj",198],["compojure.core$routes$fn__32285","invoke","core.clj",200],["metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__73621","invoke","exceptions.clj",98],["metabase.server.middleware.exceptions$catch_api_exceptions$fn__73618","invoke","exceptions.clj",86],["metabase.server.middleware.log$log_api_call$fn__77104$fn__77105$fn__77106","invoke","log.clj",209],["metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info","invokeStatic","diagnostic.clj",15],["metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info","invoke","diagnostic.clj",9],["metabase.server.middleware.log$log_api_call$fn__77104$fn__77105","invoke","log.clj",201],["toucan.db$_do_with_call_counting","invokeStatic","db.clj",216],["toucan.db$_do_with_call_counting","invoke","db.clj",209],["metabase.server.middleware.log$log_api_call$fn__77104","invoke","log.clj",200],["metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__80457","invoke","browser_cookie.clj",38],["metabase.server.middleware.security$add_security_headers$fn__55229","invoke","security.clj",148],["metabase.server.middleware.json$wrap_json_body$fn__79612","invoke","json.clj",64],["metabase.server.middleware.json$wrap_streamed_json_response$fn__79630","invoke","json.clj",98],["metabase.server.middleware.offset_paging$handle_paging$fn__55253","invoke","offset_paging.clj",42],["ring.middleware.keyword_params$wrap_keyword_params$fn__80724","invoke","keyword_params.clj",55],["ring.middleware.params$wrap_params$fn__80740","invoke","params.clj",69],["metabase.server.middleware.misc$maybe_set_site_url$fn__34379","invoke","misc.clj",59],["metabase.server.middleware.session$bind_current_user$fn__45722$fn__45723","invoke","session.clj",262],["metabase.server.middleware.session$do_with_current_user","invokeStatic","session.clj",243],["metabase.server.middleware.session$do_with_current_user","invoke","session.clj",235],["metabase.server.middleware.session$bind_current_user$fn__45722","invoke","session.clj",261],["metabase.server.middleware.session$wrap_current_user_info$fn__45709","invoke","session.clj",221],["metabase.server.middleware.session$wrap_session_id$fn__45695","invoke","session.clj",167],["metabase.server.middleware.auth$wrap_api_key$fn__65377","invoke","auth.clj",27],["ring.middleware.cookies$wrap_cookies$fn__80644","invoke","cookies.clj",216],["metabase.server.middleware.misc$add_content_type$fn__34362","invoke","misc.clj",27],["metabase.server.middleware.misc$disable_streaming_buffering$fn__34387","invoke","misc.clj",76],["ring.middleware.gzip$wrap_gzip$fn__80686","invoke","gzip.clj",86],["metabase.server.middleware.misc$bind_request$fn__34390","invoke","misc.clj",93],["metabase.server.middleware.ssl$redirect_to_https_middleware$fn__80473","invoke","ssl.clj",38],["metabase.server$async_proxy_handler$fn__76877","invoke","server.clj",73],["metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a","handle",null,-1],["org.eclipse.jetty.server.handler.HandlerWrapper","handle","HandlerWrapper.java",127],["org.eclipse.jetty.server.Server","handle","Server.java",516],["org.eclipse.jetty.server.HttpChannel","lambda$handle$1","HttpChannel.java",388],["org.eclipse.jetty.server.HttpChannel","dispatch","HttpChannel.java",633],["org.eclipse.jetty.server.HttpChannel","handle","HttpChannel.java",380],["org.eclipse.jetty.server.HttpConnection","onFillable","HttpConnection.java",277],["org.eclipse.jetty.io.AbstractConnection$ReadCallback","succeeded","AbstractConnection.java",311],["org.eclipse.jetty.io.FillInterest","fillable","FillInterest.java",105],["org.eclipse.jetty.io.ChannelEndPoint$1","run","ChannelEndPoint.java",104],["org.eclipse.jetty.util.thread.strategy.EatWhatYouKill","runTask","EatWhatYouKill.java",338],["org.eclipse.jetty.util.thread.strategy.EatWhatYouKill","doProduce","EatWhatYouKill.java",315],["org.eclipse.jetty.util.thread.strategy.EatWhatYouKill","tryProduce","EatWhatYouKill.java",173],["org.eclipse.jetty.util.thread.strategy.EatWhatYouKill","run","EatWhatYouKill.java",131],["org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread","run","ReservedThreadExecutor.java",386],["org.eclipse.jetty.util.thread.QueuedThreadPool","runJob","QueuedThreadPool.java",883],["org.eclipse.jetty.util.thread.QueuedThreadPool$Runner","run","QueuedThreadPool.java",1034],["java.lang.Thread","run","Thread.java",829]],"cause":"Invalid parameter: Card 1 does not have a template tag named nil.","data":{"type":"invalid-parameter","invalid-parameter":{"type":"date/all-options","value":"2018-03-05~2022-03-05","target":["dimension",["field",34,null]],"id":"f6dd6d27"},"allowed-parameters":null},"message":"Invalid parameter: Card 1 does not have a template tag named nil.","invalid-parameter":{"type":"date/all-options","value":"2018-03-05~2022-03-05","target":["dimension",["field",34,null]],"id":"f6dd6d27"},"allowed-parameters":null}
```

**To Reproduce**
On sample database H2:
1. Create a simple question
2. Create a dashboard and add question from above
3. Add a filter to the dashboard and connect it to question
4. Create a new user with "no self service" data access; and view only access to the collection
5. With that user open dashboard and click on a title of the question
6. Export the data
7. Open the exported data. There will be a stack trace of an error instead of correct data inside.

**Expected behavior**
I expected to download a csv file. Got error instead

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:97.0) Gecko/20100101 Firefox/97.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.13+8-Ubuntu-0ubuntu1.20.04",
    "java.vendor": "Ubuntu",
    "java.vendor.url": "https://ubuntu.com/",
    "java.version": "11.0.13",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.13+8-Ubuntu-0ubuntu1.20.04",
    "os.name": "Linux",
    "os.version": "5.13.0-27-generic",
    "user.language": "en",
    "user.timezone": "Europe/Ljubljana"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2022-02-28",
      "tag": "v0.42.2",
      "branch": "release-x.42.x",
      "hash": "d6ff494"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Relatively high. External users can't download reports directly from the dashboard. So we need to prepare exports from our side. 

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Until version 0.42.0, it would fail to access the question - #12720

**Comment by flamber** [comment]
@kanobi We definitely want users to be able to export, I was merely commenting that this is not a regressions, since you couldn't even access the question on previous versions. This is a bug, not something to do with granular permission feature.

**Comment by araujo-filipe** [comment]
I think this might be related.

On sample database H2:
1. Create a simple SQL question with a field filter
2. Create a new user with "no self service" data access; and view only access to the collection
3. With that user open the question
4. Try to select a value for the field value

The difference here is that no value will show up on the list for field filters when users have "no self service" access. This can become a bit impactful since even if a user knows which value they want, it can break because of data types (e.g. inputting "true" in a boolean field will throw a type mismatch error)

**Comment by helder-mattos** [comment]
Hello, I think I'm having the same problem. 
I have a problem concerning the data export for users that doesn't have direct query access to tables. The export action is not working (The same problem reported in following "https://discourse.metabase.com/t/user-without-query-editing-access-cannot-open-downloaded-xlsx-data-after-using-the-filters/11071")

I'm currently using Metabase v0.42.0 (de1264e release-x.42.x) 

Following is the troubleshoot log. Are there any workarounds for this to work?


```
[0e1d9b2e-0a75-45be-9d4b-ed97672ff89f] 2022-04-20T12:24:26+02:00 ERROR metabase.query-processor.middleware.catch-exceptions Erreur lors du traitement de la requête : null
{:database_id 2,
 :started_at #t "2022-04-20T10:24:25.093808Z[GMT]",
 :error_type :missing-required-permissions,
 :json_query
 {:type "query",
  :middleware
  {:ignore-cached-results? false, :process-viz-settings? true, :skip-results-metadata? true, :format-rows? false},
  :viz-settings
  {:table.pivot false,
   :table.pivot_column "gender",
   :table.cell_column "Nb de missions",
   :table.column_formatting [],
   :metabase.shared.models.visualization-settings/column-settings
   {{:metabase.shared.models.visualization-settings/field-id 2174}
    {:metabase.shared.models.visualization-settings/date-style "D/M/YYYY",
     :metabase.shared.models.visualization-settings/column-title "Date de naissance"},
    {:metabase.shared.models.visualization-settings/field-id 2171}
    {:metabase.shared.models.visualization-settings/column-title "Nationalité"},
    {:metabase.shared.models.visualization-settings/field-id 2170}
    {:metabase.shared.models.visualization-settings/column-title "Sexe"},
    {:metabase.shared.models.visualization-settings/field-id 2145}
    {:metabase.shared.models.visualization-settings/column-title "Prénom Nom Talent"},
    {:metabase.shared.models.visualization-settings/field-id 1001}
    {:metabase.shared.models.visualization-settings/column-title "ID Talent"}},
   :metabase.shared.models.visualization-settings/table-columns
   [{:metabase.shared.models.visualization-settings/table-column-name "worker_id",
     :metabase.shared.models.visualization-settings/table-column-field-ref [:field 1001 nil],
     :metabase.shared.models.visualization-settings/table-column-enabled true}
    {:metabase.shared.models.visualization-settings/table-column-name "full_name",
     :metabase.shared.models.visualization-settings/table-column-field-ref
     [:field 2145 {:join-alias "Vw User Talent"}],
     :metabase.shared.models.visualization-settings/table-column-enabled true}
    {:metabase.shared.models.visualization-settings/table-column-name "gender",
     :metabase.shared.models.visualization-settings/table-column-field-ref
     [:field 2170 {:join-alias "Vw User Talent"}],
     :metabase.shared.models.visualization-settings/table-column-enabled true}
    {:metabase.shared.models.visualization-settings/table-column-name "nationality",
     :metabase.shared.models.visualization-settings/table-column-field-ref
     [:field 2171 {:join-alias "Vw User Talent"}],
     :metabase.shared.models.visualization-settings/table-column-enabled true}
    {:metabase.shared.models.visualization-settings/table-column-name "date_of_birth",
     :metabase.shared.models.visualization-settings/table-column-field-ref
     [:field 2174 {:temporal-unit :day, :join-alias "Vw User Talent"}],
     :metabase.shared.models.visualization-settings/table-column-enabled true}
    {:metabase.shared.models.visualization-settings/table-column-name "Date de début",
     :metabase.shared.models.visualization-settings/table-column-field-ref [:aggregation 1],
     :metabase.shared.models.visualization-settings/table-column-enabled true}
    {:metabase.shared.models.visualization-settings/table-column-name "Date de fin",
     :metabase.shared.models.visualization-settings/table-column-field-ref [:aggregation 0],
     :metabase.shared.models.visualization-settings/table-column-enabled true}
    {:metabase.shared.models.visualization-settings/table-column-name "Nb de missions",
     :metabase.shared.models.visualization-settings/table-column-field-ref [:aggregation 2],
     :metabase.shared.models.visualization-settings/table-column-enabled true}]},
  :database 2,
  :query
  {:source-table 52,
   :filter ["=" ["field" 1050 nil] "option1" "option2" "option3" "option4"],
   :joins
   [{:fields
     [["field" 2145 {:join-alias "Vw User Talent"}]
      ["field" 2170 {:join-alias "Vw User Talent"}]
      ["field" 2171 {:join-alias "Vw User Talent"}]
      ["field" 2174 {:temporal-unit "day", :join-alias "Vw User Talent"}]],
     :source-table 141,
     :condition ["=" ["field" 1001 nil] ["field" 2186 {:join-alias "Vw User Talent"}]],
     :alias "Vw User Talent"}
    {:fields "none",
     :source-table 150,
     :condition ["=" ["field" 972 nil] ["field" 2779 {:join-alias "Accounting"}]],
     :alias "Accounting"}],
   :aggregation
   [["aggregation-options" ["max" ["field" 1020 nil]] {:name "Date de fin", :display-name "Date de fin"}]
    ["aggregation-options" ["min" ["field" 999 nil]] {:name "Date de début", :display-name "Date de début"}]
    ["aggregation-options" ["count"] {:name "Nb de missions", :display-name "Nb de missions"}]],
   :breakout
   [["field" 1001 nil]
    ["field" 2145 {:join-alias "Vw User Talent"}]
    ["field" 2170 {:join-alias "Vw User Talent"}]
    ["field" 2171 {:join-alias "Vw User Talent"}]
    ["field" 2174 {:temporal-unit "day", :join-alias "Vw User Talent"}]],
   :order-by [["asc" ["aggregation" 1]]]},
  :parameters
  [{:type "id",
    :value ["oe5C6Tc21S"],
    :target ["dimension" ["field" 2779 {:join-alias "Accounting"}]],
    :id "85d2f4cb"}],
  :async? true,
  :cache-ttl nil},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:29)"
  "query_processor.middleware.permissions$perms_exception.doInvoke(permissions.clj:24)"
  "query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:26)"
  "query_processor.middleware.permissions$perms_exception.invoke(permissions.clj:24)"
  "query_processor.middleware.permissions$fn__46278$check_ad_hoc_query_perms__46283$fn__46284.invoke(permissions.clj:83)"
  "query_processor.middleware.permissions$fn__46278$check_ad_hoc_query_perms__46283.invoke(permissions.clj:79)"
  "query_processor.middleware.permissions$fn__46310$check_query_permissions_STAR___46315$fn__46316.invoke(permissions.clj:98)"
  "query_processor.middleware.permissions$fn__46310$check_query_permissions_STAR___46315.invoke(permissions.clj:88)"
  "query_processor.middleware.permissions$check_query_permissions$fn__46329.invoke(permissions.clj:107)"
  "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__51010.invoke(pre_alias_aggregations.clj:40)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__46725.invoke(cumulative_aggregations.clj:60)"
  "query_processor.middleware.visualization_settings$update_viz_settings$fn__46663.invoke(visualization_settings.clj:62)"
  "query_processor.middleware.fix_bad_references$fix_bad_references_middleware$fn__50975.invoke(fix_bad_references.clj:91)"
  "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__47610.invoke(resolve_joined_fields.clj:111)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__51591.invoke(resolve_joins.clj:176)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__51133.invoke(add_implicit_joins.clj:202)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__47629.invoke(large_int_id.clj:59)"
  "query_processor.middleware.format_rows$format_rows$fn__51185.invoke(format_rows.clj:74)"
  "query_processor.middleware.add_default_temporal_unit$add_default_temporal_unit$fn__46997.invoke(add_default_temporal_unit.clj:23)"
  "query_processor.middleware.desugar$desugar$fn__46636.invoke(desugar.clj:21)"
  "query_processor.middleware.binning$update_binning_strategy$fn__39550.invoke(binning.clj:229)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__45971.invoke(resolve_fields.clj:34)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__50342.invoke(add_dimension_projections.clj:487)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__50636.invoke(add_implicit_clauses.clj:164)"
  "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__48017.invoke(upgrade_field_literals.clj:117)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__47374.invoke(add_source_metadata.clj:125)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__50887.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__48974.invoke(auto_bucket_datetimes.clj:147)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__45952.invoke(resolve_source_table.clj:45)"
  "query_processor.middleware.parameters$substitute_parameters$fn__48628.invoke(parameters.clj:111)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46025.invoke(resolve_referenced.clj:79)"
  "query_processor.middleware.expand_macros$expand_macros$fn__52249.invoke(expand_macros.clj:184)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__48407.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__51200.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__50647$fn__50652.invoke(resolve_database_and_driver.clj:35)"
  "driver$do_with_driver.invokeStatic(driver.clj:60)"
  "driver$do_with_driver.invoke(driver.clj:56)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__50647.invoke(resolve_database_and_driver.clj:34)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__46571.invoke(fetch_source_query.clj:286)"
  "query_processor.middleware.store$initialize_store$fn__46762$fn__46763.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
  "query_processor.store$do_with_store.invoke(store.clj:38)"
  "query_processor.middleware.store$initialize_store$fn__46762.invoke(store.clj:10)"
  "query_processor.middleware.validate$validate_query$fn__50982.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__50989.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__48353.invoke(add_rows_truncated.clj:35)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49645.invoke(results_metadata.clj:78)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__48371.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__50923.invoke(process_userland_query.clj:146)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__51280.invoke(catch_exceptions.clj:169)"
  "query_processor.reducible$async_qp$qp_STAR___43323$thunk__43324.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___43323$fn__43326.invoke(reducible.clj:108)"],
 :card_id nil,
 :context :csv-download,
 :error "Vous nêtes pas autorisé à exécuter cette requête.",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data
 {:type :missing-required-permissions,
  :required-permissions
  #{"/db/2/schema/mongo001/table/52/query/"
    "/db/2/schema/redshift/table/141/query/"
    "/db/2/schema/mongo001/table/150/query/"},
  :actual-permissions
  #{"/db/2/schema/redshift/table/79/"
    "/collection/23/"
    "/db/2/schema/redshift/table/576/"
    "/collection/22/"
    "/collection/19/"
    "/db/2/schema/mongo001/table/150/"},
  :permissions-error? true},
 :data {:rows [], :cols []}}
```

**Comment by kanobi** [comment]
@flamber It looks like this was resolved in 0.43 version.

**Comment by kanobi** [comment]
Ah, actually not. The exception is now put into CSV export:

![image](https://user-images.githubusercontent.com/1461037/168418810-a0aa31a9-d111-4318-8581-6daa90eebc24.png)

**Comment by kokoon** [comment]
> @kanobi We definitely want users to be able to export, I was merely commenting that this is not a regression, since you couldn't even access the question on previous versions. This is a bug, not something to do with granular permission feature.

Is this a particularly tough cookie to crack? Are you aware of any (failed) attempts to fix this?

**Comment by kokoon** [comment]
@flamber sure, I know how this works, no worries. I was mainly trying to assess if it's worth putting a few hours aside for this, if it's not a known cursed bug or something like that.

**Comment by flamber** [comment]
@kokoon Permission issues are definitely not in the "easy" class, neither are nested queries, or connection issues.

**Comment by kokoon** [comment]
@flamber what just happened with the comments? They're all bugged out now :shrug:

**Comment by kanobi** [comment]
@flamber It's weird that most comments here are hidden as spam.
Anyway, we're just trying to assess how long it would take to resolve this bug and where in the timeline of this project this fix would be planned out?

We know there are a lot of other issues that you are dealing with, so the question is more, if you have any long term plans where this would be put in the pipeline? Since this is a P2 bug there is some kind of priority relative to other 600 bugs?

We're also interested in contributing, although nobody in the team is fluent in Clojure. So we could take a look if this bug could be fixed without detailed knowledge of the whole permission and data-sandboxing system.

**Comment by flamber** [comment]
I understand this issue is annoying. But this is our issue tracker. I collapse comments, which doesn't provide more context to the actual issue. It's impossible to maintain such a big open source repo, where I read hundred of messages each day and need to keep it fairly clean, so I don't forget anything, when we address an issue.

Please use the forum for questions and troubleshooting: https://discourse.metabase.com/

**Comment by kanobi** [comment]
@flamber I proposed a fix in #23581

**Comment by kanobi** [comment]
@luizarakaki Hi, just want to check if this issue was fixed, because I tested the latest 0.46 version and it was still there.
Was it actually fixed in 0.47 version?

**Comment by luizarakaki** [comment]
Right. I mixed this issue with another similar one. This wasn't fixed.

Reopening it. I'm pretty sure this is caused because the parameter changes the query definition, internally creating a new question. When this happens, downloads won't work with no self-service.

We should be smart here and notice that the question derives from a resultset the user has access to and enable downloads.

**Comment by nachimehta** [comment]
👍 for desire for fix here. Our team is experiencing downtime because of this. Would be great to have it merged in sooner rather than later.

**Comment by kanobi** [comment]
@luizarakaki Could you check the PR that is connected with this issue? It fixed the issue for me and there was just a frontend fix with the template file. 
It's just missing the tests for the fix, which I wasn't sure how to make.

**Comment by amzar96** [comment]
Hi, I am using v0.45.2.1 and the error still happens.

```
[eeb2a0ce-1aa5-467e-bf96-2c5b2a2af709] 2023-10-20T18:22:17+08:00 ERROR metabase.server.middleware.log POST /api/card/xxx/query/xlsx 500 13.5 ms (7 DB calls) 
{:via
 [{:type clojure.lang.ExceptionInfo,
   :message "Invalid parameter: Card xxx does not have a template tag named nil.",
   :data
   {:type :invalid-parameter,
    :invalid-parameter
```

**Comment by darksciencebase** [comment]
@amzar96 This fix has been shipped for 47.1 and later versions, it is not available on 45.

**Comment by darksciencebase** [comment]
@amzar96 also, 45.2.1 is vulnerable to a critical security issue, please upgrade, you need at least 45.4.3: https://www.metabase.com/blog/security-incident-summary
